core/trino-main/src/main/java/io/trino/operator/AggregationMetrics.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.operator;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.Duration;
+import io.trino.plugin.base.metrics.DurationTiming;
+import io.trino.plugin.base.metrics.LongCount;
+import io.trino.spi.metrics.Metric;
+import io.trino.spi.metrics.Metrics;
+
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+public class AggregationMetrics
+{
+    @VisibleForTesting
+    static final String INPUT_ROWS_WITH_PARTIAL_AGGREGATION_DISABLED_METRIC_NAME = "Input rows processed without partial aggregation enabled";
+    private static final String ACCUMULATOR_TIME_METRIC_NAME = "Accumulator update wall time";
+    private static final String GROUP_BY_HASH_TIME_METRIC_NAME = "Group by hash update wall time";
+
+    private long accumulatorTimeNanos;
+    private boolean hasAccumulator;
+    private long groupByHashTimeNanos;
+    private boolean hasGroupByHash;
+    private long inputRowsProcessedWithPartialAggregationDisabled;
+    private boolean hasInputRowsProcessedWithPartialAggregationDisabled;
```

Comment:
I followed what we already have in [PageProcessorMetrics](https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/operator/project/PageProcessorMetrics.java). Additionally, `inputRowsProcessedWithPartialAggregationDisabled` was handled in a similar way. 

I assumed the reasoning was to avoid polluting the explain output with metrics that weren’t reported, as unreported metrics are irrelevant for a given operator.

Perhaps we could replace the boolean flags by checking if a given metric has a value greater than zero, and only expose it in the `AggregationMetrics::getMetrics` method in that case. WDYT?